hbase-server/src/main/java/org/apache/hadoop/hbase/security/access/SnapshotScannerHDFSAclController.java

```diff
@@ -263,10 +265,36 @@ public void postDeleteTable(ObserverContext<MasterCoprocessorEnvironment> ctx,
     }
   }
 
+  @Override
+  public void postModifyTable(ObserverContext<MasterCoprocessorEnvironment> ctx,
+      TableName tableName, TableDescriptor oldDescriptor, TableDescriptor currentDescriptor)
+      throws IOException {
+    if (checkTable(currentDescriptor, () -> "modifyTable " + tableName)
+        && !hdfsAclHelper.isTableUserScanSnapshotEnabled(oldDescriptor)) {
+      hdfsAclHelper.createTableDirectories(tableName);
+      Set<String> users = new HashSet<>();
```

Comment:
Abstract an small method here ?  I saw many places has the logic:  to pick up all the users who has the read permission on table / ns / global etc...